src/pages/iou/IOUDetailsModal.js

```diff
@@ -99,6 +99,22 @@ class IOUDetailsModal extends Component {
      * @param {String} paymentMethodType
      */
     performIOUPayment(paymentMethodType) {
+        // If it's paying with the wallet lets call the correct service for this
+        if (paymentMethodType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY) {
+            IOU.payMoneyRequestWithWallet(
+                this.props.route.params.chatReportID,
+                this.props.route.params.iouReportID,
+                this.props.iouReport.total,
+                this.props.iouReport.currency,
+                this.props.iouReport.ownerEmail,
+                this.props.iouReport.managerEmail,
+            );
+
+            // Not sure if we should navigate to the chat report here as well, but I think we should
+            Navigation.navigate(ROUTES.REPORT);
```

Comment:
I think we would like to do the same as here https://github.com/Expensify/App/blob/240eca4ce517f57ed5d6a22e6e9b6a340867aa6c/src/libs/actions/IOU.js#L223